lib/Actions.js

```diff
@@ -115,7 +172,46 @@ Actions.prototype = {
         }
 
         return vow.all([actionPromise, waitingForPage]).spread(function (result) {
-            return result;
+            return casesPromise || result;
+        });
+    },
+
+    _performCases: function (cases, parentSelector) {
+        debug('handle several cases in parallel %o', cases);
+
+        var wonCase = null;
+        var promises = cases.map(function (actions, caseNumber) {
+            var beginningPromise = this._performAction(actions[0], parentSelector);
+            return actions
+                .slice(1)
```

Comment:
if you have empty `case actions` you are f*cken mother f*cker. And even in this case everything will work like a charm. [].slice(1) returns [];